matchms/Spectrum.py

```diff
@@ -203,13 +204,22 @@ def metadata(self):
     def metadata(self, value):
         self._metadata.data = value
 
-    @property
-    def losses(self) -> Optional[Fragments]:
-        return self._losses.clone() if self._losses is not None else None
-
-    @losses.setter
-    def losses(self, value: Fragments):
-        self._losses = value
+    def losses(self, loss_mz_from, loss_mz_to) -> Optional[Fragments]:
+        precursor_mz = self.get("precursor_mz", None)
+        if precursor_mz:
+            assert isinstance(precursor_mz, (float, int)), ("Expected 'precursor_mz' to be a scalar number.",
+                                                            "Consider applying 'add_precursor_mz' filter first.")
+            peaks_mz, peaks_intensities = self.peaks.mz, self.peaks.intensities
+            losses_mz = (precursor_mz - peaks_mz)[::-1]
+            losses_intensities = peaks_intensities[::-1]
+            # Add losses which are within given boundaries
+            mask = np.where((losses_mz >= loss_mz_from)
+                            & (losses_mz <= loss_mz_to))
+            losses = Fragments(mz=losses_mz[mask],
+                               intensities=losses_intensities[mask])
+            return losses
+        logger.warning("No precursor_mz found. Consider applying 'add_precursor_mz' filter first.")
+        return None
```

Comment:
Is it possible in python to have a property that is in fact a proper function? This way, the compatibility with the `spectrum.losses` pipelines would be retained.

I'm not sure on how to incorporate the filtering here though.

Reply:
That seems like a nice solution